src/net/connector/tcp.rs

```diff
@@ -47,10 +49,13 @@ impl Connector for Direct {
     ) -> Result<Box<dyn Socket>, ConnectError> {
         info!("establishing tcp connection to {}", candidate);
 
-        let stream: Box<dyn Socket> =
-            Box::new(TcpStream::connect(candidate).await.context(Io)?);
+        backoff::future::retry(ExponentialBackoff::default(), || async {
```

Comment:
This would be better off in a separate Backoff Connector that can use any other `Connector`